sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/cloud/core/implementation/util/AzurePropertiesUtils.java

```diff
@@ -69,7 +69,11 @@ private AzurePropertiesUtils() {
         copyPropertiesIgnoreNull(source.getCredential(), target.getCredential());
 
         if (source instanceof RetryOptionsProvider && target instanceof RetryOptionsProvider) {
-            copyPropertiesIgnoreNull(((RetryOptionsProvider) source).getRetry(), ((RetryOptionsProvider) target).getRetry());
+            RetryOptionsProvider.RetryOptions sourceRetry = ((RetryOptionsProvider) source).getRetry();
+            RetryOptionsProvider.RetryOptions targetRetry = ((RetryOptionsProvider) target).getRetry();
+            copyPropertiesIgnoreNull(sourceRetry, targetRetry);
+            copyPropertiesIgnoreNull(sourceRetry.getExponential(), targetRetry.getExponential());
+            copyPropertiesIgnoreNull(sourceRetry.getFixed(), targetRetry.getFixed());
```

Comment:
Hi, @saragluna , this is a bug.
Could you please add more tests in `AzurePropertiesUtilsTests` after this PR merged?

Reply:
ok